nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/OIDCAccessResource.java

```diff
@@ -175,9 +175,13 @@ public void oidcCallback(@Context HttpServletRequest httpServletRequest, @Contex
             }
 
             // redirect to the name page
-            httpServletResponse.sendRedirect(getNiFiUri());
-        } else {
-            // remove the oidc request cookie
+            if (successfulOidcResponse.getState().getValue().contains(".")) {
+                String redirectPagePathBase64 = successfulOidcResponse.getState().getValue().substring(successfulOidcResponse.getState().getValue().lastIndexOf(".") + 1);
+                httpServletResponse.sendRedirect(getNiFiUri() + new String(Base64.getDecoder().decode(redirectPagePathBase64)));
+            } else {
```

Comment:
Change to clean code standard

Reply:
Build the url inside the if and else statements, and move the redirect call httpServletResponse.sendRedirect under the if else condition (line 184 for example)